### ISSUE TITLE
Crash in RequestSerializer

### ISSUE THREAD

**Issue by danielrhammond** [aws/aws-sdk-ios#95]
Experiencing the same crash as observed in #59 in version 2.0.12 (Via cocoapods)

````
SIGSEGV SEGV_ACCERR
Thread: Unknown Name (Crashed)
0  libobjc.A.dylib  0x0000000193f400b4 objc_retain + 20
1  [App Name] 0x000000010023f968 -[AWSXMLRequestSerializer serializeRequest:headers:parameters:] (AWSURLRequestSerialization.m:181)
2  [App Name] 0x000000010024657c __41-[AWSURLSessionManager taskWithDelegate:]_block_invoke164 (AWSURLSessionManager.m:204)
3  [App Name] 0x0000000100256370 __48-[BFTask continueWithExecutor:withSuccessBlock:]_block_invoke (BFTask.m:335)
4  [App Name] 0x0000000100255dc8 __41-[BFTask continueWithExecutor:withBlock:]_block_invoke_2 (BFTask.m:287)
5  [App Name] 0x0000000100253af4 __29+[BFExecutor defaultExecutor]_block_invoke_2 (BFExecutor.m:43)
6  [App Name] 0x0000000100253fdc -[BFExecutor execute:] (BFExecutor.m:107)
7  [App Name] 0x0000000100255d80 __41-[BFTask continueWithExecutor:withBlock:]_block_invoke (BFTask.m:284)
8  [App Name] 0x0000000100255bf8 -[BFTask continueWithExecutor:withBlock:] (BFTask.m:319)
9  [App Name] 0x0000000100256268 -[BFTask continueWithExecutor:withSuccessBlock:] (BFTask.m:331)
10 [App Name] 0x00000001002563fc -[BFTask continueWithSuccessBlock:] (BFTask.m:341)
11 [App Name] 0x0000000100245eb8 -[AWSURLSessionManager taskWithDelegate:] (AWSURLSessionManager.m:150)
12 [App Name] 0x0000000100248550 __61-[AWSURLSessionManager URLSession:task:didCompleteWithError:]_block_invoke (AWSURLSessionManager.m:403)
13 [App Name] 0x0000000100256370 __48-[BFTask continueWithExecutor:withSuccessBlock:]_block_invoke (BFTask.m:335)
14 [App Name] 0x0000000100255dc8 __41-[BFTask continueWithExecutor:withBlock:]_block_invoke_2 (BFTask.m:287)
15 [App Name] 0x0000000100253af4 __29+[BFExecutor defaultExecutor]_block_invoke_2 (BFExecutor.m:43)
16 [App Name] 0x0000000100253fdc -[BFExecutor execute:] (BFExecutor.m:107)
17 [App Name] 0x0000000100255d80 __41-[BFTask continueWithExecutor:withBlock:]_block_invoke (BFTask.m:284)
18 [App Name] 0x0000000100255bf8 -[BFTask continueWithExecutor:withBlock:] (BFTask.m:319)
19 [App Name] 0x0000000100256268 -[BFTask continueWithExecutor:withSuccessBlock:] (BFTask.m:331)
20 [App Name] 0x00000001002563fc -[BFTask continueWithSuccessBlock:] (BFTask.m:341)
21 [App Name] 0x000000010024704c -[AWSURLSessionManager URLSession:task:didCompleteWithError:] (AWSURLSessionManager.m:272)
22 CFNetwork  0x000000018334a150 __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke227 + 60
````

[Full stack trace](http://cl.ly/text/1p3r431b0g31)

**Comment by yosuke-matsuda** [comment]
We've released the AWS Mobile SDK for iOS 2.0.14. It should address the issue you encountered.

Thanks,

**Comment by danielrhammond** [comment]
@yosuke-matsuda Thanks for the feedback that it has been addressed in 2.0.14. Can confirm it works against my test case :+1:
